src/main/scala/ai/starlake/job/ingest/IngestionJob.scala

```diff
@@ -117,33 +91,34 @@ trait IngestionJob extends SparkJob {
   }
 
   protected def saveRejected(
-    errMessagesRDD: RDD[String],
-    rejectedLinesRDD: RDD[String]
+    errMessagesDS: Dataset[String],
+    rejectedLinesDS: Dataset[String]
   ): Try[Path] = {
     logger.whenDebugEnabled {
-      logger.debug(s"rejectedRDD SIZE ${errMessagesRDD.count()}")
-      errMessagesRDD.take(100).foreach(rejected => logger.debug(rejected.replaceAll("\n", "|")))
+      logger.debug(s"rejectedRDD SIZE ${errMessagesDS.count()}")
+      errMessagesDS.take(100).foreach(rejected => logger.debug(rejected.replaceAll("\n", "|")))
     }
     val domainName = domain.name
     val schemaName = schema.name
 
     val start = Timestamp.from(Instant.now())
     val formattedDate = new java.text.SimpleDateFormat("yyyyMMddHHmmss").format(start)
 
-    if (settings.comet.sinkReplayToFile && !rejectedLinesRDD.isEmpty()) {
+    if (settings.comet.sinkReplayToFile && rejectedLinesDS.limit(1).count() > 0) {
```

Comment:
a little sad to miss `isEmpty` on datasets. Is this due to back-compatiblity?

Reply:
Exactly.
For a customer you know very well. ```isEmpty``` appeared in Spark 2.4